madmom/features/tempo.py

```diff
@@ -605,6 +725,44 @@ def interval_histogram(self, activations):
         return interval_histogram_comb(activations, self.alpha,
                                        self.min_interval, self.max_interval)
 
+    def online_interval_histogram(self, activation):
+        """
+        Compute the histogram of the beat intervals using a resonating
+        comb filter bank for online mode.
+
+        Parameters
+        ----------
+        activation : numpy float
+            Beat activation for a single frame.
+
+        Returns
+        -------
+        histogram_bins : numpy array
+            Bins of the tempo histogram.
+        histogram_delays : numpy array
+            Corresponding delays [frames].
+        """
+        # copy the activation for every tau
+        copied_signal = np.full(len(self.taus), activation, dtype=np.float)
+        # append it to the comb filter matrix
+        self.combfilter_matrix.append(copied_signal)
+        # online feed backward comb filter
+        min_tau = min(self.taus)
+        for t in self.taus:
+            if len(self.combfilter_matrix) > t:
+                self.combfilter_matrix[-1][t - min_tau] += self.alpha * \
+                    self.combfilter_matrix[-1 - t][t - min_tau]
+        # retrieve maxima
+        act_max = self.combfilter_matrix[-1] == \
+            np.max(self.combfilter_matrix[-1], axis=-1)
+        # compute the max bins
+        bins = self.combfilter_matrix[-1] * act_max
+        # use a buffer to only keep bins of the last seconds
+        # shift buffer and put new bins at end of buffer
+        buffer = self._buffer(bins)
+        # build a histogram together with the intervals and return it
+        return np.sum(buffer, axis=0), np.array(self.taus)
+
```

Comment:
Note: if I do a simple measurement of the performance with `time.time()` for the whole `online_interval_histogram()` method I get an average execution time of almost exactly 1ms (on my 2013 MacBook Air).